resident/resident-service/src/main/java/io/mosip/resident/service/impl/ProxyMasterdataServiceImpl.java

```diff
@@ -513,6 +530,81 @@ public ResponseWrapper<?> getAllDynamicFieldByName(String fieldName) throws Resi
 		return responseWrapper;
 	}
 
+	@Override
+	public LocationImmediateChildrenResponseDto getImmediateChildrenByLocCode(String locationCode, List<String> languageCodes) throws ResidentServiceCheckedException {
+		List<String> cacheKeyList = languageCodes.stream()
+				.map(languageCode -> locationCode + ResidentConstants.UNDER_SCORE + languageCode)
+				.collect(Collectors.toList());
+		LocationImmediateChildrenResponseDto result =new LocationImmediateChildrenResponseDto();
+		Map<String, List<Map<String, Object>>> locations = new HashMap<>();
+		List<String> languageCodesNotCached = new ArrayList<>();
+		if (!cache.isEmpty()) {
+			cacheKeyList
+					.forEach(cacheKeyLanguage -> {
+						List<Map<String, Object>> cachedResult = cache.get(cacheKeyLanguage);
+						String languageCode = List.of(cacheKeyLanguage.split(ResidentConstants.UNDER_SCORE)).get(1);
+						if (cachedResult != null) {
+							locations.put(languageCode, cachedResult);
+						} else {
+							languageCodesNotCached.add(languageCode);
+						}
+					});
+		}
+		if(cache.isEmpty()){
+			languageCodesNotCached.addAll(languageCodes);
+		}
+		if(!languageCodesNotCached.isEmpty()) {
+			LocationImmediateChildrenResponseDto responseDto = getImmediateChildrenByLocCodeAndLanguageCodes(locationCode, languageCodes);
```

Comment:
pass languageCodesNotCached instead of  languageCodes here to the API.

Reply:
yes done.